scipy/stats/tests/test_stats.py

```diff
@@ -7949,50 +7949,41 @@ def test_no_args_gh20661(self):
+                             # > sumlog(x1)
+                             [("fisher", [.01, .2, .3], 0.02156175132483465),
+                              # > sumz(x1)
+                              ("stouffer", [.01, .2, .3], 0.01651203260896294),
+                              # > sumz(x2)
+                              ("stouffer", [.5, .5, .5], 0.5),
+                              # need to work out how to generate this
+                              ("pearson", [.01, .2, .3], 0.02213),
+                              # > minimump(x1)
+                              ("tippett", [.01, .2, .3], 0.02970100000000002),
+                              # need to work out how to generate this
+                              # (mean of fisher and pearson)
+                              ("mudholkar_george", [.01, .2, .3], 0.019462)])
+    def test_reference_values(self, xp, method, pvalues, expected):
+        res = stats.combine_pvalues(xp.asarray(pvalues), method=method)
+        xp_assert_close(res.pvalue, xp.asarray(expected))
+
+    @pytest.mark.parametrize("weights,expected",
+                             # Same as no weights
+                             [([1., 1., 1.], 0.01651203260896294),
+                              # > weights = c(1., 4., 9.)
+                              # > sumz(x1, weights=weights)
+                              ([1., 4., 9.], 0.1464422142261314)])
+    def test_weighted_stouffer(self, xp, weights, expected):
         pvalues = xp.asarray([.01, .2, .3])
-        Z, p = stats.combine_pvalues(pvalues, method='stouffer',
-                                     weights=xp.ones(3, dtype=pvalues.dtype))
-        xp_assert_close(p, xp.asarray(0.01651), rtol=1e-3)
-
-    def test_weighted_stouffer2(self, xp):
-        Z, p = stats.combine_pvalues(xp.asarray([.01, .2, .3]), method='stouffer',
-                                     weights=xp.asarray([1., 4., 9.]))
-        xp_assert_close(p, xp.asarray(0.1464), rtol=1e-3)
-
-    def test_pearson(self, xp):
-        Z, p = stats.combine_pvalues(xp.asarray([.01, .2, .3]), method='pearson')
-        xp_assert_close(p, xp.asarray(0.02213), rtol=1e-3)
-
-    def test_tippett(self, xp):
-        Z, p = stats.combine_pvalues(xp.asarray([.01, .2, .3]), method='tippett')
-        xp_assert_close(p, xp.asarray(0.0297), rtol=1e-4)
-
-    def test_mudholkar_george(self, xp):
-        Z, p = stats.combine_pvalues(xp.asarray([.1, .1, .1]),
-                                     method='mudholkar_george')
-        xp_assert_close(p, xp.asarray(0.019462), rtol=1e-4)
-
-    def test_mudholkar_george_equal_fisher_pearson_average(self, xp):
-        Z, p = stats.combine_pvalues(xp.asarray([.01, .2, .3]),
-                                     method='mudholkar_george')
-        Z_f, p_f = stats.combine_pvalues(xp.asarray([.01, .2, .3]), method='fisher')
-        Z_p, p_p = stats.combine_pvalues(xp.asarray([.01, .2, .3]), method='pearson')
-        xp_assert_close(0.5 * (Z_f+Z_p), Z, rtol=1e-4)
```

Comment:
This test seems redundant, why do we need to test a property if with the same input they are tested against correct reference values

Reply:
Could turn this into a property-based test by using random p-values, though.